fpmsyncd/routesync.cpp

```diff
@@ -886,6 +1004,105 @@ bool RouteSync::getIfName(int if_index, char *if_name, size_t name_len)
     return true;
 }
 
+/*
+ * getNextHopList() - parses next hop list attached to route_obj
+ * @arg route_obj     (input) Netlink route object
+ * @arg gw_list       (output) comma-separated list of NH IP gateways
+ * @arg mpls_list     (output) comma-separated list of NH MPLS info
+ * @arg intf_list     (output) comma-separated list of NH interfaces
+ *
+ * Return bool true=Success false=Failure
+ */
+bool RouteSync::getNextHopList(struct rtnl_route *route_obj, string& gw_list,
+                               string& mpls_list, string& intf_list)
+{
+    bool mpls_found = false;
+
+    for (int i = 0; i < rtnl_route_get_nnexthops(route_obj); i++)
+    {
+        struct rtnl_nexthop *nexthop = rtnl_route_nexthop_n(route_obj, i);
+        struct nl_addr *addr = NULL;
+
+        /* RTA_GATEWAY is NH gateway info for IP routes only */
+        if ((addr = rtnl_route_nh_get_gateway(nexthop)))
```

Comment:
can we make this as one condition so that there is only one set of code .

` if ((addr = rtnl_route_nh_get_gateway(nexthop)) || (addr = rtnl_route_nh_get_via(nexthop)))`

Reply:
Done.  function is updated to have only one condition for mulitple attrs.